src/optimalcontrol/include/iDynTree/Constraint.h

```diff
@@ -15,59 +15,169 @@
 
```

Comment:
Only as a matter of style. Since here we''re using indented namespaces, in order to avoid too many indentation levels, wouldn't be more appropriate a forward declaration on the top of headers files? For example:

```cpp
namespace iDynTree {
    class MatrixDynSize;
    namespace optimalcontrol {
        class Constraint;
    }
}

class iDynTree::optimalcontrol::Constraint : ...
...

```

Reply:
Not sure. This allows to save space only if the namespace name is lower than 4 characters. Right?

Reply:
Sorry got it later :grin:. Actually I don't have a firm opinion on this. Do we have a defined style for ``iDynTree``?